### DISCUSSION TITLE
DeprecationWarning for a couple of versions before removing public methods

### DISCUSSION THREAD

**Discussion by bitplane** [agronholm/anyio#610] [Ideas]
Hi

Just tracking down build errors and found the new version released yesterday! It broke starlette's test client.

I know it's my fault for not freezing deps, but any chance of a `DeprecationWarning` in future?

https://github.com/encode/starlette/issues/2262

**Comment by agronholm** [comment]
The offending function had already been moved way back in v3.0.0 along with all the other thread-related functions, but curiously it was left out of the changelog – a human error. I have often added such warnings (most recently in `create_memory_object_stream()`, but this one slipped through the cracks. That said, anyone casually upgrading a SemVer project to the next major version without testing is just asking for trouble.
